### ISSUE TITLE
possible bootnode option issue?

### ISSUE THREAD

**Issue by unsol** [ethereumjs/ethereumjs-client#103]
Okay, this one is confusing to me.

EthereumJS 0.0.5 is running on an On-Demand AWS t2.large (2vCPU, 8GiB memory, 50GiB SSD with 150 IOPS and 46GiB free space remaining) EC2 using Ubuntu 18.04.2 LTS.

After the #102 issue, I wiped everything, and started from scratch. Cloned the repo, npm i, npm run build, then `./cli.js --network goerli`

First time through, I get this
```INFO [03-06|16:46:05] Data directory: /home/ubuntu/Library/Ethereum/goerli/ethereumjs/chaindata                                        
INFO [03-06|16:46:05] Initializing Ethereumjs client...                              
INFO [03-06|16:46:05] Fast sync mode        
INFO [03-06|16:46:05] Connecting to network: goerli 
INFO [03-06|16:46:05] Latest local block: number=0 td=1 hash=bf7e331f...             
INFO [03-06|16:46:05] Synchronizing blockchain...                           
INFO [03-06|16:46:05] Started rlpx server.                                       
INFO [03-06|16:46:05] Started libp2p server.                                         
INFO [03-06|16:46:05] Listener up transport=rlpx url=enode://7e2eb24aec46930e30d033e7846ea6008885276afc493fc2813e2282342086dd48e44212357c9bada99c3593c30d9c4d18cb6747c237c
47fbd5e602af99fc72b@[::]:30303              
INFO [03-06|16:46:06] Listener up transport=libp2p url=/ip4/127.0.0.1/tcp/50580/ws/ipfs/QmdthbSnnoAN8sdyZGyajgbG6xtwwgjAdAT5X1Z2Mnm2ns       
INFO [03-06|16:46:06] Started eth service.                                  
WARN [03-06|16:46:07] Peer error: Error: Not enough data for decode id=46add44b address=94.237.54.114:45214 transport=rlpx protocols=eth inbound=true
ERROR [03-06|16:46:09] uncaughtException: Cannot read property 'toString' of null    
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)                                                                            
    at RLPx.emit (events.js:197:13)                                         
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs
-devp2p/lib/rlpx/index.js:202:34)                                                    
    at Peer.emit (events.js:197:13)                                                      at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16) 
    at Socket.emit (events.js:197:13)                                                    at addChunk (_stream_readable.js:288:12)            
    at readableAddChunk (_stream_readable.js:269:11)                        
    at Socket.Readable.push (_stream_readable.js:224:10)                                 at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)        
ERROR [03-06|16:46:09] uncaughtException: Cannot read property 'toString' of null   
TypeError: Cannot read property 'toString' of null                                  
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)                                                                           
    at RLPx.emit (events.js:197:13)                                                 
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs-devp2p/lib/rlpx/index.js:202:34)                                                   
    at Peer.emit (events.js:197:13)                                                 
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)                                                    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)   
    at Socket.Readable.push (_stream_readable.js:224:10)                      
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
```

The peer error warning had me thinking maybe it had something to do with the bootnode, but I wasn't quite sure how to specify an alternate, so I haphazardly started EthereumJS with `./cli.js --network goerli --bootnode`
That time, the client began to sync normally:
```INFO [03-06|16:49:50] Data directory: /home/ubuntu/Library/Ethereum/goerli/ethereumjs
/chaindata                                                                           INFO [03-06|16:49:50] Initializing Ethereumjs client...                          
INFO [03-06|16:49:50] Fast sync mode              
INFO [03-06|16:49:50] Connecting to network: goerli                                  INFO [03-06|16:49:50] Latest local block: number=0 td=1 hash=bf7e331f...
INFO [03-06|16:49:50] Synchronizing blockchain...                                   
INFO [03-06|16:49:50] Started rlpx server.                                           INFO [03-06|16:49:50] Started libp2p server.
INFO [03-06|16:49:50] Listener up transport=rlpx url=enode://1cc0603410bd19103da11e24
713af840db250d231d9a2b37731ab30dd999951043da6bcbede34d574c528a3f64d90a86adedb7827cbef
330ea22bad4dda18cc9@[::]:30303                                                       INFO [03-06|16:49:52] Listener up transport=libp2p url=/ip4/127.0.0.1/tcp/50580/ws/ip
fs/QmPQbRE9QyzRwmdRXJD3iGwD2Hpy6SGx2mvENFfF9zSNBs                                   
INFO [03-06|16:49:52] Started eth service.                                          
INFO [03-06|16:49:55] Imported blocks count=128 number=1 hash=8f5bab21... peers=2
INFO [03-06|16:49:55] Imported blocks count=128 number=129 hash=831d8538... peers=2
INFO [03-06|16:49:55] Imported blocks count=128 number=257 hash=80ed202f... peers=2
INFO [03-06|16:49:56] Imported blocks count=128 number=385 hash=e849bd62... peers=2
INFO [03-06|16:49:56] Imported blocks count=128 number=513 hash=ca5a77d0... peers=2  
INFO [03-06|16:49:56] Imported blocks count=128 number=641 hash=64660f44... peers=2
INFO [03-06|16:49:57] Imported blocks count=128 number=769 hash=06759da0... peers=2
INFO [03-06|16:49:57] Imported blocks count=128 number=897 hash=2da235c0... peers=2  
INFO [03-06|16:49:57] Imported blocks count=128 number=1025 hash=6597fea3... peers=2
INFO [03-06|16:49:58] Imported blocks count=128 number=1153 hash=fdac3b24... peers=2
ERROR [03-06|16:49:58] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)
    at RLPx.emit (events.js:197:13)
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs
-devp2p/lib/rlpx/index.js:202:34)
    at Peer.emit (events.js:197:13)
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
ERROR [03-06|16:49:58] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)
    at RLPx.emit (events.js:197:13)
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs
-devp2p/lib/rlpx/index.js:202:34)
    at Peer.emit (events.js:197:13)
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
INFO [03-06|16:49:58] Imported blocks count=128 number=1281 hash=2044236c... peers=2
INFO [03-06|16:49:59] Imported blocks count=128 number=1409 hash=22cf6b9f... peers=2
INFO [03-06|16:49:59] Imported blocks count=128 number=1537 hash=8caf0fc2... peers=2
INFO [03-06|16:49:59] Imported blocks count=128 number=1665 hash=d737fcc5... peers=2
INFO [03-06|16:50:00] Imported blocks count=128 number=1793 hash=82c049d6... peers=2
INFO [03-06|16:50:00] Imported blocks count=128 number=1921 hash=2bb6b20c... peers=2
INFO [03-06|16:50:00] Imported blocks count=128 number=2049 hash=7e5c885c... peers=2
^CINFO [03-06|16:50:00] Caught interrupt signal. Shutting down...
```

Curious, I started EthereumJS again with `./cli.js --network goerli`
```INFO [03-06|16:50:06] Data directory: /home/ubuntu/Library/Ethereum/goerli/ethereumjs
/chaindata                                                                           
INFO [03-06|16:50:06] Initializing Ethereumjs client...                              
INFO [03-06|16:50:06] Fast sync mode                                                 
INFO [03-06|16:50:06] Connecting to network: goerli
INFO [03-06|16:50:06] Latest local block: number=2304 td=4609 hash=f14df8e0...
INFO [03-06|16:50:06] Synchronizing blockchain...
INFO [03-06|16:50:06] Started rlpx server.
INFO [03-06|16:50:06] Started libp2p server.
INFO [03-06|16:50:06] Listener up transport=rlpx url=enode://35482cd9b97dcd35cdf8150$
4d3f34da2db5fe667c46340cca7f5472d006cf5db23599dac2caf035b48470959f5aa8e2c697a808b9fd$
eb90a7723c1f8880f9a@[::]:30303
INFO [03-06|16:50:08] Listener up transport=libp2p url=/ip4/127.0.0.1/tcp/50580/ws/i$
fs/QmdJLmiHMFJthAFYDfsKdBkfrWcdpw7LZePkwnJfNtjHRF
INFO [03-06|16:50:08] Started eth service.
ERROR [03-06|16:50:08] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.$
s:213:34)
    at RLPx.emit (events.js:197:13)
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumj$
-devp2p/lib/rlpx/index.js:202:34)
    at Peer.emit (events.js:197:13)
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_module$
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
ERROR [03-06|16:50:08] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.$
s:213:34)
    at RLPx.emit (events.js:197:13)
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumj$
-devp2p/lib/rlpx/index.js:202:34)
    at Peer.emit (events.js:197:13)
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_module$
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
ERROR [03-06|16:50:08] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.$
s:213:34)
    at RLPx.emit (events.js:197:13)
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumj$
-devp2p/lib/rlpx/index.js:202:34)
    at Peer.emit (events.js:197:13)
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_module$
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)
    at Socket.emit (events.js:197:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)
ERROR [03-06|16:50:08] uncaughtException: Cannot read property 'toString' of null    
TypeError: Cannot read property 'toString' of null                                   
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)                                                                            
    at RLPx.emit (events.js:197:13)                                                  
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs
-devp2p/lib/rlpx/index.js:202:34)                                                    
    at Peer.emit (events.js:197:13)                                                  
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)                                          
    at Socket.emit (events.js:197:13)                                                
    at addChunk (_stream_readable.js:288:12)                                         
    at readableAddChunk (_stream_readable.js:269:11)                                 
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)  
ERROR [03-06|16:50:08] uncaughtException: Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at RLPx.rlpx.on (/home/ubuntu/ethereumjs-client-0.0.5/lib/net/server/rlpxserver.j
s:213:34)                                                                            
    at RLPx.emit (events.js:197:13)                                                  
    at Peer.peer.on.err (/home/ubuntu/ethereumjs-client-0.0.5/node_modules/ethereumjs
-devp2p/lib/rlpx/index.js:202:34)                                                    
    at Peer.emit (events.js:197:13)              
    at Socket.Peer._socket.on.data (/home/ubuntu/ethereumjs-client-0.0.5/node_modules
/ethereumjs-devp2p/lib/rlpx/peer.js:112:16)                                      
    at Socket.emit (events.js:197:13)             
    at addChunk (_stream_readable.js:288:12)                                         
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:145:17)         
WARN [03-06|16:50:11] Peer error: Error: Not enough data for decode id=46add44b addre
ss=94.237.54.114:33092 transport=rlpx protocols=eth inbound=true
INFO [03-06|16:50:11] Imported blocks count=128 number=2305 hash=c034ff06... peers=2
```

Now isn't that odd? That's the whole trace right there. It warns, syncs one block, then goes straight back to awaiting a command. No messages, or anything.

I tried it a few times, and had the same thing happen each time. For whatever reason, specifying an empty bootnode option is the only way I can connect to Goerli right now.

**Comment by unsol** [comment]
You know what, I totally lied right there. My apologies! I extracted from the v0.0.5 tar'd release, and encountered those issues.

I wiped everything, actually cloned the repo this time, and things are synching without the empty bootnode option. I think that ghost crash was #96. 

I still don't understand why it wasn't synching without an empty bootnode option though. Did the address change at some point?

**Comment by vpulim** [comment]
Ok, phew 😅  

Yeah, that error looked like the one that was fixed in #96. I think it was pure luck that it worked with the `--bootnode` option because I'm looking right now at the code, and that is not a valid option and gets totally ignored. The way you are supposed to specify a bootnode is via this option: `--transports rlpx:bootnodes=52.74.57.123:30303`

**Comment by unsol** [comment]
Weird! Well, guess I'm just under-caffeinated, and lucky. I'll close this up then. Sorry for making you sweat. Haha! Thanks again!
